# Code along for Probability 

## Import libries

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import scipy as stat
from zipfile import ZipFile
from functools import reduce
import os

## Load files

### Extract the zip


In [2]:
path="data/archive.zip"
extratedfile="data/new_files"

# extracted the files to path "data/new_files"
with ZipFile (path, 'r') as f:
    f.extractall(extratedfile)

### Load to dataframe

In [3]:
# let load the extracted files into dictionary
dataframes={}
for file in os.listdir(extratedfile):
    if file.endswith("csv"):
        file_path= os.path.join(extratedfile,file)
        dataframes[file]= pd.read_csv(file_path)

In [4]:
# let view the list of our dataset
for name, df in dataframes.items():
    print(name)

olist_customers_dataset.csv
olist_geolocation_dataset.csv
olist_orders_dataset.csv
olist_order_items_dataset.csv
olist_order_payments_dataset.csv
olist_order_reviews_dataset.csv
olist_products_dataset.csv
olist_sellers_dataset.csv
product_category_name_translation.csv


### Let load each dataset into appropriate variable

In [5]:
# Let load customer
customers=pd.read_csv(f'{extratedfile}/olist_customers_dataset.csv')
customers.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

In [6]:
# Let load geolocation
geolocation=pd.read_csv(f'{extratedfile}/olist_geolocation_dataset.csv')
geolocation.head()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                         1037       -23.545621       -46.639292   
1                         1046       -23.546081       -46.644820   
2                         1046       -23.546129       -46.642951   
3                         1041       -23.544392       -46.639499   
4                         1035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP

In [7]:
# Let load orders
orders=pd.read_csv(f'{extratedfile}/olist_orders_dataset.csv')
orders.head()


order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [8]:
# Let load order items
order_items=pd.read_csv(f'{extratedfile}/olist_order_items_dataset.csv')
order_items.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

In [9]:
# Let load order payment
order_payments=pd.read_csv(f'{extratedfile}/olist_order_payments_dataset.csv')
order_payments.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [10]:
# Let load Orders review
order_reviews=pd.read_csv(f'{extratedfile}/olist_order_reviews_dataset.csv')
order_reviews.head(2)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   

   review_score review_comment_title review_comment_message  \
0             4                  NaN                    NaN   
1             5                  NaN                    NaN   

  review_creation_date review_answer_timestamp  
0  2018-01-18 00:00:00     2018-01-18 21:46:59  
1  2018-03-10 00:00:00     2018-03-11 03:05:13

In [11]:
# let load products
products=pd.read_csv(f'{extratedfile}/olist_products_dataset.csv')
products.head()

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

In [12]:
# Let load selllers
sellers=pd.read_csv(f'{extratedfile}/olist_sellers_dataset.csv')
sellers.head()

seller_id  seller_zip_code_prefix  \
0  3442f8959a84dea7ee197c632cb2df15                   13023   
1  d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2  ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3  c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4  51a04a8a6bdcb23deccc82b0b80742cf                   12914   

         seller_city seller_state  
0           campinas           SP  
1         mogi guacu           SP  
2     rio de janeiro           RJ  
3          sao paulo           SP  
4  braganca paulista           SP

In [13]:
# let load product category_name
product_category=pd.read_csv(f'{extratedfile}/product_category_name_translation.csv')

## Check for column interset for all the datas

In [14]:
# to check all intersection let get all the column heads
all_file_columns={name: set(df.columns) for name,df in dataframes.items()}
column_interset= reduce(set.intersection,all_file_columns.values())
print("The intersect column is:", column_interset)

The intersect column is: set()


## Merge the data set 

In [15]:
# check all the columns for each dataset
for name, file in all_file_columns.items():
    print(f"{name}: {list(file)}")

olist_customers_dataset.csv: ['customer_unique_id', 'customer_zip_code_prefix', 'customer_state', 'customer_id', 'customer_city']
olist_geolocation_dataset.csv: ['geolocation_city', 'geolocation_zip_code_prefix', 'geolocation_lat', 'geolocation_state', 'geolocation_lng']
olist_orders_dataset.csv: ['order_delivered_carrier_date', 'order_approved_at', 'order_purchase_timestamp', 'order_id', 'order_status', 'order_estimated_delivery_date', 'customer_id', 'order_delivered_customer_date']
olist_order_items_dataset.csv: ['order_item_id', 'order_id', 'product_id', 'shipping_limit_date', 'price', 'freight_value', 'seller_id']
olist_order_payments_dataset.csv: ['payment_value', 'order_id', 'payment_type', 'payment_installments', 'payment_sequential']
olist_order_reviews_dataset.csv: ['order_id', 'review_comment_message', 'review_answer_timestamp', 'review_id', 'review_comment_title', 'review_score', 'review_creation_date']
olist_products_dataset.csv: ['product_description_lenght', 'product_phot

In [16]:
# merging the file if required
# Merge customers and orders on customer_id
orders_customers= pd.merge(orders,customers, on="customer_id", how="right")
orders_customers.head(2)

order_id                       customer_id  \
0  00e7ee1b050b8499577073aeb2a297a1  06b8999e2fba1a1fbc88172c00ba8bc7   
1  29150127e6685892b6eab3eec79f59c7  18955e83d337fd6b2def6b18a428ac77   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-05-16 15:05:35  2017-05-16 15:22:12   
1    delivered      2018-01-12 20:48:24  2018-01-12 20:58:32   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-05-23 10:47:57           2017-05-25 10:35:35   
1          2018-01-15 17:14:59           2018-01-29 12:41:19   

  order_estimated_delivery_date                customer_unique_id  \
0           2017-06-05 00:00:00  861eff4711a542e4b93843c6dd7febb0   
1           2018-02-06 00:00:00  290c77bc529b7ac935b93aa66c333dc3   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP

In [17]:
# orders_customers(merge of customers and orders) + Order_payment
orders_payments = pd.merge(orders_customers, order_payments, on="order_id", how="left")
orders_payments.head(2)

order_id                       customer_id  \
0  00e7ee1b050b8499577073aeb2a297a1  06b8999e2fba1a1fbc88172c00ba8bc7   
1  29150127e6685892b6eab3eec79f59c7  18955e83d337fd6b2def6b18a428ac77   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-05-16 15:05:35  2017-05-16 15:22:12   
1    delivered      2018-01-12 20:48:24  2018-01-12 20:58:32   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-05-23 10:47:57           2017-05-25 10:35:35   
1          2018-01-15 17:14:59           2018-01-29 12:41:19   

  order_estimated_delivery_date                customer_unique_id  \
0           2017-06-05 00:00:00  861eff4711a542e4b93843c6dd7febb0   
1           2018-02-06 00:00:00  290c77bc529b7ac935b93aa66c333dc3   

   customer_zip_code_prefix          customer_city customer_state  \
0                     14409                 franca             SP   
1                      9790  sao bernardo do campo             SP   

   payment_sequential payment_type  payment_installments  payment_value  
0                 1.0  credit_card                   2.0         146.87  
1                 1.0  credit_card                   8.0         335.48

In [18]:
# Order_payment (the last merge )+ order_review
orders_reviews = pd.merge(orders_payments, order_reviews, on="order_id", how="left")
order_reviews.head(2)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   

   review_score review_comment_title review_comment_message  \
0             4                  NaN                    NaN   
1             5                  NaN                    NaN   

  review_creation_date review_answer_timestamp  
0  2018-01-18 00:00:00     2018-01-18 21:46:59  
1  2018-03-10 00:00:00     2018-03-11 03:05:13

In [19]:
# orders_reviews(the last merge )+ order_item
orders_items= pd.merge(order_reviews, order_items, on="order_id", how="left")
orders_items.head(2)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   

   review_score review_comment_title review_comment_message  \
0             4                  NaN                    NaN   
1             4                  NaN                    NaN   

  review_creation_date review_answer_timestamp  order_item_id  \
0  2018-01-18 00:00:00     2018-01-18 21:46:59            1.0   
1  2018-01-18 00:00:00     2018-01-18 21:46:59            2.0   

                         product_id                         seller_id  \
0  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7   
1  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7   

   shipping_limit_date  price  freight_value  
0  2018-01-18 15:47:59  185.0          13.63  
1  2018-01-18 15:47:59  185.0          13.63

In [20]:
# orders_items(the last merge )+ product
orders_products = pd.merge(orders_items, products, on="product_id", how="left")
orders_products.head(2)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   

   review_score review_comment_title review_comment_message  \
0             4                  NaN                    NaN   
1             4                  NaN                    NaN   

  review_creation_date review_answer_timestamp  order_item_id  \
0  2018-01-18 00:00:00     2018-01-18 21:46:59            1.0   
1  2018-01-18 00:00:00     2018-01-18 21:46:59            2.0   

                         product_id                         seller_id  ...  \
0  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   
1  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   

   price  freight_value  product_category_name product_name_lenght  \
0  185.0          13.63          esporte_lazer                42.0   
1  185.0          13.63          esporte_lazer                42.0   

   product_description_lenght  product_photos_qty  product_weight_g  \
0                       858.0                 1.0            1300.0   
1                       858.0                 1.0            1300.0   

   product_length_cm  product_height_cm  product_width_cm  
0               30.0               30.0              35.0  
1               30.0               30.0              35.0  

[2 rows x 21 columns]

In [21]:
# orders_products (last merge) + Product Category Translation
orders_categories = pd.merge(orders_products, product_category, on="product_category_name", how="left")
orders_products.head(2)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   

   review_score review_comment_title review_comment_message  \
0             4                  NaN                    NaN   
1             4                  NaN                    NaN   

  review_creation_date review_answer_timestamp  order_item_id  \
0  2018-01-18 00:00:00     2018-01-18 21:46:59            1.0   
1  2018-01-18 00:00:00     2018-01-18 21:46:59            2.0   

                         product_id                         seller_id  ...  \
0  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   
1  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   

   price  freight_value  product_category_name product_name_lenght  \
0  185.0          13.63          esporte_lazer                42.0   
1  185.0          13.63          esporte_lazer                42.0   

   product_description_lenght  product_photos_qty  product_weight_g  \
0                       858.0                 1.0            1300.0   
1                       858.0                 1.0            1300.0   

   product_length_cm  product_height_cm  product_width_cm  
0               30.0               30.0              35.0  
1               30.0               30.0              35.0  

[2 rows x 21 columns]

In [22]:
orders_sellers = pd.merge(orders_categories, sellers, on="seller_id", how="left")
orders_sellers.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
2  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
3  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
4  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   

   review_score review_comment_title                 review_comment_message  \
0             4                  NaN                                    NaN   
1             4                  NaN                                    NaN   
2             5                  NaN                                    NaN   
3             5                  NaN                                    NaN   
4             5                  NaN  Recebi bem antes do prazo estipulado.   

  review_creation_date review_answer_timestamp  order_item_id  \
0  2018-01-18 00:00:00     2018-01-18 21:46:59            1.0   
1  2018-01-18 00:00:00     2018-01-18 21:46:59            2.0   
2  2018-03-10 00:00:00     2018-03-11 03:05:13            1.0   
3  2018-02-17 00:00:00     2018-02-18 14:36:24            1.0   
4  2017-04-21 00:00:00     2017-04-21 22:02:06            1.0   

                         product_id                         seller_id  ...  \
0  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   
1  fd25ab760bfbba13c198fa3b4f1a0cd3  6d803cb79cc31c41c4c789a75933b3c7  ...   
2  be0dbdc3d67d55727a65d4cd696ca73c  8e6d7754bc7e0f22c96d255ebda59eba  ...   
3  d1c427060a0f73f6b889a5c7c61f2ac4  a1043bafd471dff536d0c462352beb48  ...   
4  52c80cedd4e90108bf4fa6a206ef6b03  a1043bafd471dff536d0c462352beb48  ...   

  product_description_lenght  product_photos_qty  product_weight_g  \
0                      858.0                 1.0            1300.0   
1                      858.0                 1.0            1300.0   
2                      493.0                 1.0             245.0   
3                     1893.0                 1.0            6550.0   
4                     2188.0                 2.0            7650.0   

  product_length_cm  product_height_cm  product_width_cm  \
0              30.0               30.0              35.0   
1              30.0               30.0              35.0   
2              19.0               14.0              14.0   
3              20.0               20.0              20.0   
4              20.0               20.0              20.0   

   product_category_name_english  seller_zip_code_prefix  \
0                 sports_leisure                 14600.0   
1                 sports_leisure                 14600.0   
2          computers_accessories                 12233.0   
3          computers_accessories                 37175.0   
4                   garden_tools                 37175.0   

            seller_city  seller_state  
0  sao joaquim da barra            SP  
1  sao joaquim da barra            SP  
2   sao jose dos campos            SP  
3              ilicinea            MG  
4              ilicinea            MG  

[5 rows x 25 columns]

In [23]:
# Customers + Geolocation (via zip prefix)
geo_unique = geolocation.groupby("geolocation_zip_code_prefix").first().reset_index()
final_df= pd.merge(orders_sellers,geo_unique,left_on="customer_zip_code_prefix", right_on="geolocation_zip_code_prefix", how="left" )
final_df.head()

KeyError: 'customer_zip_code_prefix'

## Probability

#### 1. Probability of an Event
Question: What is the probability that a randomly chosen order received a 5-star review?

In [ ]:
# probability probability that a randomly chosen order received a 5-star review
FiveStar_review= order_reviews['review_score']==5
p_5star= FiveStar_review.mean()
print(f"probability of 5 stars = {round(p_5star *100,2)}%")

probability of 5 stars = 57.78%


**Interpretation**

probability that a randomly chosen order received a 5-star review is 57.78%. This show that on average, customer is happy with the service

#### 2. Complementary Probability

Concept: 
𝑃(A complement) = 1−P(A)

Question: What is the probability that a review is not 5 stars?

In [ ]:
# since we know prob. of getting 5 star, so 𝑃(A complement) = 1−P(A)
p_not_5star= 1-p_5star
print(f"probability of not 5 stars = {round(p_not_5star *100,2)}%")

probability of not 5 stars = 42.22%


**Interpretation**

On average customer given less than 5 star is below average, which mean customers are satisfied with the service and give 5 start reviews more.

#### 3. Joint Probability

Concept: P(A∩B)

Question: What is the probability that an order was paid by credit card and received a 5-star review?

In [ ]:
# let merge the order_payment and Order_review table together
order_review_payment= pd.merge(order_reviews,order_payments,on="order_id", how="left")

# let get P(credit _card and 5star)
Credit_card_5star= ((order_review_payment['payment_type']=="credit_card")&(order_review_payment['review_score']==5)).sum()
p_5star_Credit_card= Credit_card_5star/ len(order_review_payment)
print(f"probability of (credit _card and 5star) = {round(p_5star_Credit_card *100,2)}%")

probability of (credit _card and 5star) = 42.76%


In [ ]:
(order_review_payment['payment_type']=="Credit_card").count()
# (order_review_payment['review_score']==5).sum()
# len(order_review_payment)

103678

**Interpretation**

#### 4. Conditional Probability

Concept:

`𝑃(𝐴∣𝐵)=𝑃(𝐴∩𝐵)/𝑃(𝐵)` 
                
`𝑃(𝐴∣𝐵)= Favourable_outcomes / Total_credit_card_outcomes`


Question: What is the probability that a review is 5 stars given that the payment was made by credit card?
Do it for all:
𝑃(5∣credit_card)

P(5∣voucher)

P(5∣boleto)

**Interpretation**

#### 5. Bayes’ Rule 

Concept:

`P(A∣B )= P(B∣A)⋅P(A) / P(B)`

A = payment

B = review =5

`P(payment | review=5) = P(review=5 | payment).p(payment) / p(review =5)`


Question: Given that an order received a 5-star review, what is the probability that it was paid with each payment method (credit_card, boleto, voucher)?


Among only the 5-star reviews, how are payment methods distributed?

**Interpretation**

## **Expectation and Variance**

**Question1**

What is the average delivery time (expectation) and variance (spread/consistency) of deliveries?

**Interpretation**

**Question2**

What is the expected customer review score and how much do reviews vary?

**Interpretation**

## **Sampling & Estimation**

**Question 1:**

 Take a random sample of 50 orders

**Question 2:**

 Take a random sample of 200 orders

**Interpretation**

## **Law of Large Numbers (LLN)**

- The Law of Large Numbers says,

"As the sample size increases, the sample mean → converges to the true population mean"

- Let us experiment a bit...

## **Central Limit Theorem (CLT)**

**Question:**

Collect 100 samples of review_scores, take the mean. Repeat the same process a thousand times.
Then take all the 1000 means use it to plot a histogram and lets see what the shape or distribution would look like.


## **One-Sample Test**

**Question1**

In the Olist e-commerce dataset, customer reviews are scored from 1 (worst) to 5 (best). A score of 3 is considered neutral.
Using the review data, test whether the average customer review score is significantly different from 3 (neutral)

## **Two-Sample t-test**

**Question1**

- Extract the payment_value for orders paid with Credit Card and Boleto.

  - Perform a two-sample t-test and interpret the results

     - What are the average spends for each group?

     - What is the p-value?

     - Based on the p-value, do we reject or fail to reject the null hypothesis?

     - What does this mean in practical terms about customer spending behavior?
